### PR TITLE
[add] Added support for iOS in @ionic-native/nfc

### PR DESCRIPTION
**Pull request by lehrenberg**

### CODE REVIEW

**Comment by lehrenberg** [comment]
Thanks @stormwin for the feedback! I changed what you suggested and pushed the code.

**Comment by janpio** [comment]
@lehrenberg You misunderstood @stormwin, no need to delete `package-lock.json`. You only should not change it. Deleting it is as bad as modifying it.

Awesome first PR though!

**Comment by lehrenberg** [comment]
Thanks @janpio ! I added the current package-lock again. Hope that was correct this time.

**Comment by janpio** [comment]
Yep, https://github.com/ionic-team/ionic-native/pull/2050/files now only lists your actual code changes.

File: src/@ionic-native/plugins/nfc/index.ts [hunk 1]
```diff
@@ -49,6 +49,21 @@ declare let window: any;
...
+   * Starts the NFCNDEFReaderSession allowing iOS to scan NFC tags.
+   * @param onSuccess
+   * @param onFailure
+   * @returns {Observable<any>}
+   */
+  @Cordova({
+    observable: true,
+    successIndex: 0,
+    errorIndex: 3,
+    clearFunction: 'invalidateSession',
```
Comment (janpio) [inline src/@ionic-native/plugins/nfc/index.ts]: will this work although the function is not defined in the wrapper?

Comment (lehrenberg) [reply src/@ionic-native/plugins/nfc/index.ts]: Apparently, yes. @stormwin said the ionic wrapper would be smart enough to call this function.

**Comment by Jackthomsonn** [comment]
What’s the status on this guys?

**Comment by janpio** [comment]
This is a PR ready to be reviewed, merged and eventually released.

**Comment by Jackthomsonn** [comment]
Are there any workarounds for the time being? I cloned this repo added the above built the modules and replaced my ionic/native/nfc with the updated dist files but the app kept crashing after declaring beginSession()..

**Comment by joshbabb** [comment]
@Jackthomsonn, could you provide any warnings or errors from the commandline or console?

What platform are you on? Version? etc. etc.

**Comment by joshbabb** [comment]
For reference: (see [the official documentation](https://github.com/ionic-team/ionic-native/blob/master/DEVELOPER.md))

1. Clone/fork [ionic-native](https://github.com/lehrenberg/ionic-native/)

2. Install dependencies

  `npm install`

3. Build Ionic Native

  `npm run build`

4. Manually copy the resulting `\dist` to `..\YourProject\node_modules\@ionic-native\dist`

**Comment by Jackthomsonn** [comment]
@joshbabb Sorry for the delayed response, I had tried doing all of what you said above but every time I tap the card to my phone the app just crashes. I can't even debug the app as it crashes everything. I will put my code below for reference, maybe I'm using the beginSession wrong. I also noticed this has been merged in so I am using the latest release

```
this.nfc.beginSession().subscribe(() => {
  this.nfc.addNdefListener((data) => {
      this.navCtrl.push(Page, { test: data })
    })
  })
})
```

The 'Page' reference I am using in the navctrl is defined also at the top and is used elsewhere.

**Comment by chelinho139** [comment]
@Jackthomsonn did you figure it out? status on this issue. thanks!

**Comment by Jackthomsonn** [comment]
Hello @chelinho139 - I have still not resolved this issue. I am not sure if it is to do with the NFC tags I am using. I have encoded them with an Android app so IOS could read them but I’m not sure if there are some extra steps I am missing.

**Comment by eefano** [comment]
Hello, any updates on this, please?

**Comment by UsherNet** [comment]
Hi, I'm looking for an NFC solution for Ionic (otherwise I can only release the Android version and have to write a full native option for iOS).  Is this commit going to make it to release?

Thanks

- Paul.

**Comment by jrajba** [comment]
This solution works form me.

`readTagIos(){
        this.nfc.beginSession((data) => {}, 
            (err) => {}
            ).subscribe((event)=> {
                this.nfc.addNdefListener((event)=>{}
                ).subscribe((event)=>{
                   
                })
            })
    }`

**Pull request merged**